domain/src/main/kotlin/no/nav/su/se/bakover/domain/vedtak/Stønadsvedtak.kt

```diff
@@ -566,3 +779,18 @@ fun List<VedtakSomKanRevurderes>.lagTidslinje(periode: Periode): Tidslinje<Vedta
             objekter = it,
         )
     }
+
+private fun Dokumenttilstand?.setDokumentTilstandBasertPåBehandlingHvisNull(b: Behandling): Dokumenttilstand =
+    when (this) {
+        null -> when (b.skalSendeVedtaksbrev()) {
+            true -> Dokumenttilstand.IKKE_GENERERT_ENDA
+            false -> Dokumenttilstand.SKAL_IKKE_GENERERE
+        }
+
+        else -> this
+    }
+
+private fun IverksattRevurdering.dokumenttilstandForBrevvalg(): Dokumenttilstand = when (this.brevvalgRevurdering) {
+    is BrevvalgRevurdering.Valgt.IkkeSendBrev -> Dokumenttilstand.SKAL_IKKE_GENERERE
+    is BrevvalgRevurdering.Valgt.SendBrev -> Dokumenttilstand.IKKE_GENERERT_ENDA
+}
```

Comment:
jeg la på dette. var litt :thinkies: når jeg bare så på koden her i Pr'en

Reply:
Endret denne til å gjelde på nivå Behandling. Den vil ikke gjelde for avvist klage eller avslått søknadsbehandling siden vi generer brev med en gang i de tilfellene. Har lagt på litt guards.